src/OpenTracing/LogUtils.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace OpenTracing\LogUtils;
+
+use Exception;
+use InvalidArgumentException;
+use OpenTracing\LogField;
+use Throwable;
+
+function interleavedKVToFieldsConverter(array $keyValueFields)
+{
+    $fields = [];
+
+    foreach ($keyValueFields as $key => $value) {
+        if (!is_string($key)) {
+            throw new InvalidArgumentException(sprintf("Non-string key, got %s of type %s", $key, gettype($key)));
+        }
+
+        switch (true) {
+            case is_bool($value):
```

Comment:
Can't find it anymore, read it a long time ago. I now that function calls have a bigger overhead in PHP than in any other language so anything trivial that avoids a function call is significantly faster if called often enough